subsys/bluetooth/host/smp.c

```diff
@@ -5750,16 +5750,19 @@ void bt_smp_update_keys(struct bt_conn *conn)
 	 * security level upon encryption
 	 */
 	switch (smp->method) {
+	case LE_SC_OOB:
+	case LEGACY_OOB:
+		conn->le.keys->flags |= BT_KEYS_OOB;
+		/* fallthrough */
 	case PASSKEY_DISPLAY:
```

Comment:
Seeing as how the discussion in [#46923](https://github.com/zephyrproject-rtos/zephyr/pull/46923) is coming along, maybe we should differentiate between these two. We can then check just `BT_KEYS_LE_SC_OOB` for the `LESC` permission.

Its possible that a 128B permission is still needed to be (re-)added, to cover the Legacy OOB with 128B entropy key for LE Audio though.

Reply:
Could you please check the current revision?

Reply:
The above mentioned issue is solved by the le-audio security check that you implemented. I think this is the best solution too!